vim25/soap/client.go

```diff
@@ -133,31 +133,35 @@ func ParseURL(s string) (*url.URL, error) {
 }
 
 func NewClient(u *url.URL, insecure bool) *Client {
+	var t *http.Transport
+
+	if d, ok := http.DefaultTransport.(*http.Transport); ok {
+		t = d.Clone()
+	} else {
+		t = new(http.Transport)
+	}
+
+	return newClientWithTransport(u, insecure, t)
+}
+
+func newClientWithTransport(u *url.URL, insecure bool, t *http.Transport) *Client {
 	c := Client{
 		u: u,
 		k: insecure,
 		d: newDebug(),
+		t: t,
 
 		Types: types.TypeFunc(),
 	}
 
-	// Initialize http.RoundTripper on client, so we can customize it below
-	if t, ok := http.DefaultTransport.(*http.Transport); ok {
-		c.t = &http.Transport{
-			Proxy:                 t.Proxy,
-			DialContext:           t.DialContext,
-			MaxIdleConns:          t.MaxIdleConns,
-			IdleConnTimeout:       t.IdleConnTimeout,
-			TLSHandshakeTimeout:   t.TLSHandshakeTimeout,
-			ExpectContinueTimeout: t.ExpectContinueTimeout,
+	c.hosts = make(map[string]string)
+	if c.k {
+		if c.t.TLSClientConfig == nil {
```

Comment:
hmm, not seeing that. `NewClient` always passes a non-nil `t`. And when called via `newServiceClientWithTransport`, `Client.t` will also be non-nil.

Reply:
Right, in its current form that's true, but if someone adds a function that passes in nil to this function, it may happen. I was speaking more from a defensive context, but feel free to ignore.

Reply:
ok, will leave as-is for now